src/ui/components/SecondaryToolbox/react-devtools/components/HooksRenderer.tsx

```diff
@@ -0,0 +1,164 @@
+import { MappedLocation, ObjectId, PauseId, Value as ProtocolValue } from "@replayio/protocol";
+import { MouseEvent, ReactNode, useContext } from "react";
+
+import Expandable from "replay-next/components/Expandable";
+import Icon from "replay-next/components/Icon";
+import KeyValueRenderer from "replay-next/components/inspector/KeyValueRendererWithContextMenu";
+import { InspectorContext } from "replay-next/src/contexts/InspectorContext";
+import { objectCache } from "replay-next/src/suspense/ObjectPreviews";
+import { ReplayClientInterface } from "shared/client/types";
+import { hookNameCache } from "ui/components/SecondaryToolbox/react-devtools/suspense/hookNameCache";
+import { findProtocolObjectProperty } from "ui/components/SecondaryToolbox/react-devtools/utils/findProtocolObjectProperty";
+import { findProtocolObjectPropertyValue } from "ui/components/SecondaryToolbox/react-devtools/utils/findProtocolObjectPropertyValue";
+import { getReplayLocationFromReactDevToolsSourceSuspends } from "ui/components/SecondaryToolbox/react-devtools/utils/getReplayLocationFromReactDevToolsSourceSuspends";
+
+import styles from "./HooksRenderer.module.css";
+
+export function HooksRenderer({
+  depth = 0,
+  objectId,
+  pauseId,
+  replayClient,
+}: {
+  depth?: number;
+  objectId: ObjectId;
+  pauseId: PauseId;
+  replayClient: ReplayClientInterface;
+}) {
+  const { inspectFunctionDefinition } = useContext(InspectorContext);
+
+  const hookObject = objectCache.read(replayClient, pauseId, `${objectId}`, "canOverflow");
+
+  let hookName = findProtocolObjectPropertyValue<string>(hookObject, "name") || "Anonymous";
+  const subHooksProperty = findProtocolObjectProperty(hookObject, "subHooks");
+  const valueProperty = findProtocolObjectProperty(hookObject, "value");
+
+  const subHooks = subHooksProperty?.object
+    ? objectCache.read(replayClient, pauseId, `${subHooksProperty.object}`, "canOverflow")
```

Comment:
:question: Suspense question I'm still a bit confused on. We've got a whole series of `someCache.read()` calls in here. Doesn't that mean the component will re-suspend N times, at least on the first render?  Does it make sense to do these in parallel?

Reply:
Yes, but we can't parallelize the calls in this case (unfortunately) because we have to first fetch the hook object to read it's "subHooks" property (if it has one) to them fetch it. I _think_ we might be able to avoid re-fetching the sub-hooks if our cache pre-caches the object previews but I'm not sure.